docs/deployment/setup_gardener.md

```diff
@@ -1,44 +1,412 @@
+
+Check the [controller registration documentation](https://github.com/gardener/gardener/blob/master/docs/extensions/controllerregistration.md) for more details.
+
+Now, the classical way of installing `ControllerDeployment` and `ControllerRegistration` is to prepare the resources and apply them to the virtual Garden cluster. Please note, the `ControllerDeployment` contains a base64 encoded archive of extension's helm chart.
+
+For example, to deploy the [provider Openstack](https://github.com/gardener/gardener-extension-provider-openstack) extension, the [example file](https://github.com/gardener/gardener-extension-provider-openstack/blob/master/example/controller-registration.yaml) can be used.
+
+To simplify this procedure, extensions can be registered through `Extension` resources in the runtime cluster. The Gardener Operator will take care of deploying the required resources to the virtual Garden cluster.
+In addition, this approach allows the extensions and their admission controllers to be deployed to the runtime cluster as well. There, they can provide additional functionality like creating DNS records or `BackupBuckets`.
+More details can be found in the [Extension Resource documentation](https://github.com/gardener/gardener/blob/master/docs/concepts/operator.md#extension-resource).
+
+![extensions](./content/extensions.png)
+
+### Cloud Profile
+
+In order to host any `Shoot` cluster, at least one `CloudProfile` is required. The resource describes supported Kubernetes and OS versions, as well as infrastructure capabilities like regions and their availability zones.
+
+Hence, the next step is to craft a `CloudProfile` for each infrastructure provider and combine it with the information where the operating system images can be found. The resulting `CloudProfile` resource has to be deployed to the virtual Garden cluster.
+
+The following list provides the high-level steps to build a `CloudProfile`:
+
+- add at least one supported minor version to `.spec.kubernetes.versions[]`.
+- add at least one region including the availability zone to `.spec.regions[]`
+- add at least one operating system to `.spec.machineImages[]` and define at least on supported version.
+- add one or several machine types to `.spec.machineTypes[]`
+- specify a `.spec.providerConfig` according to the extension's documentation. Typically, this includes a list of `MachineImages`. Garden Linux, for example, publishes images and provides a list of location as part of the release notes. In case of private infrastructure, copying the images might be required.
+
+Here is a simplified example manifest:
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: CloudProfile
+metadata:
+  name: openstack-example
+spec:
+  caBundle: |
```

Comment:
I think it will be nice to add short explanation of each field that is not mentioned before or is not obvious 
```suggestion
  caBundle: | # CA Bundle installed on all nodes of the shoot clusters using this CloudProfile
```